### ISSUE TITLE
could not open static file

### ISSUE THREAD

**Issue by Ventje23** [SickChill/SickChill#5968]
### Before submitting your issue:

Enable debug logging in SickChill settings, reproduce the error (be sure to disable after the bug is fixed)

Branch/Commit: master / c0061b1fa23e91d3acf3a7817612c6da23ef2928
OS: Synology
Browser: Chrome
What you did: updated to latest commit
What happened: getting several errors:
"could not open static file volume1/@appstore/sickchill/var/SickBeard/gui/slick/cache/images/xxxxxx.poster.jpg 
and 
volume1/@appstore/sickchill/var/SickBeard/gui/slick/cache/images//thumbnails/xxxxxx.poster.jpg

xxxxxx are several numbers e.g. 345947; 26858;  366171; 

Also at the restart of Sickchill (after the update) a whole list of the next error appeared (looks like an error for every show I have in Sickchill):
volume1/@appstore/sickchill/var/SickBeard/gui/slick/cache/images/xxxxxx/baner.jpg

When I switch to the main page i first get to see the sickchill banners or all show and after a couple of seconds the expected banners excluded the shows for which I get the afore mentioned error.

What you expected:
Logs:
```
PASTE LOGS HERE
```

**Comment by miigotu** [comment]
Are the images there?

**Comment by Ventje23** [comment]
Yes, except for like three show. Also I just found out that when I select a show I get the next two errors:
could not open static file volume1/@appstore/sickchill/var/SickBeard/gui/slick/cache/images/xxxxxx.poster.jpg
and
volume1/@appstore/sickchill/var/SickBeard/gui/slick/cache/images//thumbnails/xxxxxx.poster.jpg

whith xxxxxx the number of that specifica show. In the show screen I get to see the images tho.

**Comment by Navino16** [comment]
I experiment the same issue for all show who have a 404 error from tvdb api when trying to get images (#5967).

**Comment by miigotu** [comment]
What is the symlink path for that location without the @ in it @Ventje23 
@Navino16 the 404 log line is not an error, just shows tvdb doesnt have that image

**Comment by Ventje23** [comment]
hi,

don't really get what you mean, sorry. But i just found out something else interesting:

According to Sickhill config settings the cache folder is:
'/volume1/@appstore/sickchill/var/gui/slick/cache'
when I browse the directories with Putty I can access this folder indeed filled with jpg's + the thumbnail folder

When I go to directory 'volume1/@appstore/sickchill/var/SickBeard/gui/slick/' which is mentioned in the error, there is no directory 'cache' !!

So it looks like Sickchill is somehow trying to look in another/wrong directory??
(The directory SickBeard is put between '/var/' and '/gui/')

**Comment by miigotu** [comment]
Stop it and try setting the cavhe_dir in config.ini to the full path to the cache folder

**Comment by Ventje23** [comment]
Hi,

checked the value in config.ini: 'cache_dir = gui/slick/cache'

to what value do I set it?

It seems to me that based on the value of 'cache_dir' the cache is set to:
'/volume1/@appstore/sickchill/var/gui/slick/cache'

so it somewhere concatenates '/volume1/@appstore/sickchill/' with 'gui/slick/cache'
which is correct, all the jpg's are there.
Looks like this value it ot set in config.ini.

Sickchill tho generates an error by generating a cache folder named: 'volume1/@appstore/sickchill/var/SickBeard/gui/slick/cache/'

so it looks like Sickchill concatenates 'volume1/@appstore/sickchill/var/SickBeard/'
with the value of cache_dir.

I suspect that setting the value of cache_dir to the whole path ends up in an even weirder path for the cache.

But maybe you can explain where it goes wrong

**Comment by miigotu** [comment]
There is a direct path to that folder without the @ in it on that device. It's a symlink, and it is what is causing the issue.

**Comment by Ventje23** [comment]
hi,

so (not being too familiar with Linux) I went to: /volume1/@appstore/sickchill/var/SickBeard/gui/slick
pwd - P gives: /volume1/@appstore/sickchill/var/SickBeard/gui/slick

and:

/volume1/@appstore/sickchill/var/gui/slick/cache
pwd -P gives: /volume1/@appstore/sickchill/var/gui/slick/cache

So it seems no symlink is used...

**Comment by miigotu** [comment]
@Ventje23 what is the output of:
```bash
ls -l /var/packages/sickchill
```

**Comment by potestus** [comment]
Me too
The folder cache doesn't seem to be there. 

AA 2020-02-12 09:32:32 ERROR Could not open static file u'/share/HDA_DATA/.qpkg/SickChill/SickChill/gui/slick/cache/images/350068.banner.jpg'
AA 2020-02-12 09:32:32 ERROR Could not open static file u'/share/HDA_DATA/.qpkg/SickChill/SickChill/gui/slick/cache/images/350068.poster.jpg'
AA 2020-02-12 09:32:32 ERROR Could not open static file u'/share/HDA_DATA/.qpkg/SickChill/SickChill/gui/slick/cache/images/thumbnails/350068.poster.jpg'

**Comment by cb1ocker** [comment]
Same thing here running SickChill Version: v2020.02.11-3 on Windows.

AA 2020-02-12 07:25:54 ERROR    Could not open static file u'C:\\Program Files (x86)\\SickChill\\SickChill\\gui\\slick\\cache\\images\\350071.banner.jpg'

This path exists, but there is no cache folder in it: C:\Program Files (x86)\SickChill\SickChill\gui\slick\ 

Checking config.ini I see cache_dir as:
cache_dir = gui\slick\cache

Looking in my SickChill install folders I see the cache is actually located in:

C:\Program Files (x86)\SickChill\Data\gui\slick\cache

Stopped SickChill, changed config.ini to point at full path of existing cache: 
cache_dir = C:\Program Files (x86)\SickChill\Data\gui\slick\cache

Started SickChill and I still get the same errors referencing the path that doesn't exist:

2020-02-12 08:23:59 ERROR    Could not open static file u'C:\\Program Files (x86)\\SickChill\\SickChill\\gui\\slick\\cache\\images\\350071.banner.jpg'

**Comment by Ventje23** [comment]
Hi miigotu,

output of ls -l /var/packages/sickchill:

-rwxrwxrwx    1 root     root         54462 Oct 19  2018 INFO
drwxrwxrwx    2 root     root          4096 Oct 17  2018 WIZARD_UIFILES
drwxrwxrwx    2 root     root          4096 Oct 17  2018 conf
-rw-r--r--    1 root     root             0 Feb 10 19:25 enabled
lrwxrwxrwx    1 root     root            32 Oct 19  2018 etc -> /usr/syno/etc/packages/sickchill
drwxrwxrwx    2 root     root          4096 Jun 29  2018 scripts
lrwxrwxrwx    1 root     root            28 Oct 19  2018 target -> /volume1/@appstore/sickchill

**Comment by miigotu** [comment]
@Ventje23 and `/usr/syno/etc/packages/sickchill` ?
@cb1ocker I assume that file is not there, correct? The log line can be ignored if so, and I will see about suppressing it.
@potestus create the cache in `u'/share/HDA_DATA/.qpkg/SickChill/SickChill/gui/slick/cache', and force a full update of all shows to redownload the cache images. I feel like that may be a permissions issue, since it is moved inside a code directory now.

**Comment by Ventje23** [comment]
ls -l /usr/syno/etc/packages/sickchill gives: 

-rw-r--r--    1 root     root            18 Oct 19  2018 installer-variables

**Comment by miigotu** [comment]
I updated the formatting of your comment above and now see the path is:
`/var/packages/sickchill/target`
So stop SC, edit config.ini and set cache_dir to `/var/packages/sickchill/target/gui/slick/cache`

**Comment by tsuihak** [comment]
@miigotu 

Thank you for help. I have installed on my freenas as a jail and somehow mine was like this: cache_dir = gui\slick\cache. 
I changed the config.ini as below and solved the problem.
cache_dir = gui/slick/cache

**Comment by Ventje23** [comment]
@miigotu,

i changed the setting. At startup of SickChill no errors. BUT, after I do a full refresh of a show I get the static file errors for banner and poster and the thumbnail/poster 

so something like this:
volume1/@appstore/sickchill/var/SickBeard/gui/slick/cache/images/xxxxxx.poster.jpg

but still there is SickBeard in the path and that path doesn't contain the cache

**Comment by miigotu** [comment]
@Ventje23 the exact path is `/var/packages/sickchill/target/gui/slick/cache`
Nothing else. Put that path in config.ini
Create that folder manually if you have to.
Some files are missing, do a mass update after fixing the path.

Even after that, this "error" will show for files that are missing, because it isn't an error it's just saying a file isn't found.

**Comment by Ventje23** [comment]
@miigotu,

so yes I definitely in config.ini set `cache_dir = /var/packages/sickchill/target/gui/slick/cache`
checked the value in Help & Info: `/var/packages/sickchill/target/gui/slick/cache`

did a mass update and I am still getting the next type of errors:

volume1/@appstore/sickchill/var/SickBeard/gui/slick/cache/images/xxxxxx.banner.jpg

like:
```
volume1/@appstore/sickchill/var/SickBeard/gui/slick/cache/images/349271.banner.jpg
volume1/@appstore/sickchill/var/SickBeard/gui/slick/cache/images/thumbnails/349271.poster.jpg
```
I made two screen caps:

![image](https://user-images.githubusercontent.com/59983825/74562350-e6786900-4f6a-11ea-93e3-dc0da675922f.png)


![image](https://user-images.githubusercontent.com/59983825/74562379-fa23cf80-4f6a-11ea-959c-3cdf014fd589.png)

**Comment by miigotu** [comment]
No, all of this is because I moved the cache folder and don't quite understand how syno is not working correctly with it. Look at your path for the cache folder and thumbnails folders at the top of those screenshots, the slashes are wrong

**Comment by Ventje23** [comment]
Hi miigotu,

First of all I want (and need) to thank you for your patience and help!!

I just checked the paths and folder names and found out that a folder `\\slick\\cache` was created. 
And the images above are from that folder. So I removed that folder and its contents and started over.

Once again made sure that in config.ini cache_dir is set to '/var/packages/sickchill/target/gui/slick/cache`

checked Help & info for the value of cache: `/var/packages/sickchill/target/gui/slick/cache`
and did a mass update. And in the end of that mass update I get static file error messages for all the shows.

Here an image of the logfile:

![image](https://user-images.githubusercontent.com/59983825/74586739-05b6db00-4feb-11ea-9064-c278fe54b4c3.png)

**Comment by miigotu** [comment]
Try the develop branch

**Comment by Ventje23** [comment]
Hi,

this mornng I was ready to try the develop branche when I noticed there were some new updates, when I checked them I saw that the first one was about the force cache dir to `gui/slick/cache` and the log_dir in `data_dir/Logs`.
So I decided to try these updates first before switching to the develop branch.
(commit: 1f124b31b16303a511c86717f11dd28b9cbf7b46 and  version: v2020.02.15-3)

So, after the update I noticed that the cache_dir is now set to: `/volume1/@appstore/sickchill/var/SickBeard/gui/slick/cache`

After a mass update and trying the different shows I noticed that  static file errors aren't generated!
Also I noticed that the folder `SickBeard` appeared in the cache_dir path.
The before generated errors were referring to the now set cahce_dir. 
So it looks like the problem is fixed :=)

So many thanks for your help and fixing this issue.
Also thank you for all the work you're doing on Sickchill.
